src/yunohost/tools.py

```diff
@@ -555,3 +556,22 @@ def tools_diagnosis(auth, private=False):
         diagnosis['private']['domains'] = domain_list(auth)['domains']
 
     return diagnosis
+
+
+def tools_portavailable(port):
+    """
+    Check availability of a local port
+
+    Keyword argument:
+        port -- Port to check
+
+    """
+    try:
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        s.settimeout(1)
+        s.connect(("localhost", int(port)))
```

Comment:
Should we test to connect to localhost or to the main ip ?
May be adding a host optionnal argument could be an idea.

For the openvpn apps, in some case I need to test it on another ip. I am agree it's not a common case.

What about ipv6 ? And if a service is down ?